src/Illuminate/Collections/LazyCollection.php

```diff
@@ -42,17 +42,18 @@ class LazyCollection implements CanBeEscapedWhenCastToString, Enumerable
      */
     public function __construct($source = null)
     {
-        if ($source instanceof Closure || $source instanceof self) {
-            $this->source = $source;
-        } elseif (is_null($source)) {
-            $this->source = static::empty();
-        } elseif ($source instanceof Generator) {
-            throw new InvalidArgumentException(
+        $this->source = match (true) {
+            $source instanceof Closure,
+            $source instanceof self => $source,
+
+            is_null($source) => static::empty(),
```

Comment:
You can avoid using a function call:
```suggestion
            $source === null => static::empty(),
```

Reply:
`is_null()` is optimized by the JIT compiler, and has no difference to comparing to `null`:

https://3v4l.org/7R0iF/vld

It really doesn't make a difference which to choose. I'd stick to whichever is more common on the codebase

Note: I'm not saying it should be reverted to `is_null()`, just that there is no execution penalty in case `is_null()` is more idiomatic on the codebase.

Reply:
> I'd stick to whichever is more common on the codebase

If the most common is to use `=== null`, let's stick to it.

Readability is a very subjective argument, and using hidden subject ("is generally considered..." by whom? In which survey?) usually doesn't help to make a point.

Also, both links make a point that there is no difference in using any of the alternatives. 

The first comment on the documentation page is from 16 years ago, before PHP had a JIT compiler. And the top answer on SO links to that comment.

Take a look at the comment from 2 years ago, on the same docs page, that benchmarks it in PHP 8.1. 

I consider the difference to be within a margin of error. But if the argument is that "This topic has been previously discussed in PHP documentation", then using `is_null()`, since PHP 8.1, is marginally faster after 1.000.000.000 (a billion) iterations.

Which again, to me, should not be the sole reason to stick to any style. 

If `=== null` is more common on Laravel's codebase, then `=== null` is the best choice.

Both my comments were to address common misconceptions, or subjective arguments, which, in my opinion, should not be the basis to justify any decision-making.

Reply:
After a quick search on `laravel/framework`'s `src` directory:

- `is_null()` is found 747 times (with or without a negation/`!` before it)
- `=== null` is found 77 times
- `!== null` is found 42 times
- `null ===` is found 1 time
- `null !==` was not found

As you are the PR's author, use whichever you want, there is no difference in performance, nor in the generated byte-code.

But I guess, considering the number of contributors of the codebase, it is safe to say that more people wouldn't agree - or simply wouldn't care - about subjective arguments such as "Readability" and "generally considered more direct and explicit".